edgexfoundry/device-usb-camera
# feat: SetFps v4l2 + ffmpeg command

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

Note, any commands executed need to include the query parameter `PathIndex` with the value set to the number representing the video path stream intended. For example: `http://localhost:59982/api/v3/url?PathIndex=0`. Also it is recommended to use the updated postman collection.
1. Execute the `FrameRateFormat` api to see supported fps values. This shows all fps values for each resolution for each type of encoding. This can be useful for determining the values to set the camera fps.
2. Execute the `DataFormat` api to see the supported fps values for the current video format and encoding.
3. Execute the `SetFrameRate` api and see no errors.The fps is represented in a time per frame format, meaning the denominator divided by the numerator yields the fps value. I did this to maintain consistency with the internal data structures. You can leave the numerator out completely, and the denominator will represent the fps value.
4. Execute the `GetFrameRate` api and see that the fps match.
5. Test the video stream and/or raw device stream to see if the changes apply. Certain video players seem to reset the fps value themselves, so it may not be viable to see the fps value of the /dev/video stream reflected in them.

## New Dependency Instructions (If applicable)
